src/ObjectiveFunctions.jl

```diff
@@ -41,20 +42,20 @@ immutable LossTransform{T,L<:Loss} <: AbstractLossTransform{T}
     target::SumNode{T,1}
     output::OutputNode{T,1}
 
-    function LossTransform(loss::Loss, nin::Int)
+    function (::Type{LossTransform{T, L}}){T, L <: Loss}(loss::Loss, nin::Int)
         input = InputNode(T, nin)
         target = InputNode(T, nin)
         output = OutputNode(T, 1)
         grad(output)[1] = one(T)  # ∂L/∂L == 1
-        new(loss, nin, input, target, output)
+        new{T, L}(loss, nin, input, target, output)
     end
 end
 
 LossTransform{L<:Loss}(loss::L, nin::Int) = LossTransform{Float64, L}(loss, nin)
 
 # input and target are pre-populated... compute the output value as: ∑ loss(targetᵢ, inputᵢ)
 function transform!(lt::LossTransform)
-    lt.output.val[1] = sumvalue(lt.loss, value(lt.target), input_value(lt))
+    lt.output.val[1] = value(lt.loss, value(lt.target), input_value(lt), AvgMode.Sum())
```

Comment:
I changed this based on https://github.com/JuliaML/LossFunctions.jl/commit/774014d95fff62fbe439d8b08ae023e5cdb57aec#diff-f2fc3674e12bf71a707cb4988076d39eR9